src/programs/print_stack_statistics/print_stack_statistics.cpp

```diff
@@ -0,0 +1,80 @@
+#include "../../core/core_headers.h"
+
+class
+PrintStackStatistics : public MyApp
+{
+
+	public:
+
+	bool DoCalculation();
+	void DoInteractiveUserInput();
+
+	private:
+};
+
+
+
+IMPLEMENT_APP(PrintStackStatistics)
+
+// override the DoInteractiveUserInput
+
+void PrintStackStatistics::DoInteractiveUserInput()
+{
+
+	int new_z_size = 1;
+
+	UserInput *my_input = new UserInput("PrintStackStatistics", 1.0);
+
+	std::string input_filename_one	=		my_input->GetFilenameFromUser("Input image file name", "Filename of stack to print statistics for", "input_stack1.mrc", true );
+
+
+	delete my_input;
+
+	my_current_job.Reset(2);
+	my_current_job.ManualSetArguments("t", input_filename_one.c_str());
+
+}
+
+// override the do calculation method which will be what is actually run..
+
+bool PrintStackStatistics::DoCalculation()
+{
+	std::string	input_filename_one						= my_current_job.arguments[0].ReturnStringArgument();
+
+	ImageFile my_input_file_one(input_filename_one,false);
+
+	Image my_image_one;
+	float image_sigma;
+	float image_average;
+	float average_sigma = 0.0f;
```

Comment:
With very large stacks, to avoid rounding errors, wouldn't it be better to use doubles for accumulating the sums and computing averages/sigmas?